### ISSUE TITLE
Documentation for an Agent

### ISSUE THREAD

**Issue by pkriens** [sigoden/aichat#1062]
I am trying to use the agent facility with a RAG. However, I am totally lost in how to set it up. So far I've found:
```
    ./
       functions/
           agents.txt                              a file with the names of the agents 
           agents/
              <myagent>/                        an agent dir
                 index.yaml                       a configuration file but unknown what you can do here
                 <docs>                            mentioned in `documents`
       agents/                                     data dir?
            <myagent>/
                   rag.yaml                       hard coded name!
                   ???
```
A simple example agent (the todo from the documentation?) and one with a RAG setup would be HIGHLY appreciated. 

I was surprised that the settup did not follow the normal ./agents/<myagent>/ setup?

Anyway, thanks for this project. Works awfully nice.

**Comment by ntack93** [comment]
Same. The Agents feature simply doesn’t work.

On Dec 19, 2024, at 9:41 AM, Peter Kriens ***@***.***> wrote:

﻿

I am trying to use the agent facility with a RAG. However, I am totally lost in how to set it up. So far I've found:

    ./
       functions/
           agents.txt                              a file with the names of the agents
           agents/
              <myagent>/                        an agent dir
                 index.yaml                       a configuration file but unknown what you can do here
       agents/                                     data dir?
            <myagent>/
                   ???


A simple example agent (the todo from the documentation?) and one with a RAG setup would be HIGHLY appreciated.

I was surprised that the settup did not follow the normal ./agents// setup?

Anyway, thanks for this project. Works awfully nice.

**Comment by pkriens** [comment]
I got it to work now. I found out that the `documents` needs to be local to the index.yaml file. Then I found out that the rag was stored in `./agents/<myagent>/rag.yaml`. I overwrite this with a rag I made in the normal way. that seems to work.

![image](https://github.com/user-attachments/assets/f65f0f3a-e9c0-42b9-b927-6a0a5981c481)

But it would really help if there was a guide. There seems some odd decisions in how the configuration is laid out or I am missing something.

**Comment by sigoden** [comment]
All functions related to LLM tools and agents are handled by [llm-functions](https://github.com/sigoden/llm-functions).

We provide a [demo](https://github.com/sigoden/llm-functions/tree/main/agents/demo) agent as an example.

The agent has only one RAG, which is automatically created from the `index.yaml > documents` field. The agent does not support creating RAGs in other ways and does not support creating multiple RAGs.

**Comment by pkriens** [comment]
thanks for the quick reply. However:

* A small Agent guide that points to this documentation would help
* The link to the llm repo should be clearly pointed out?
* I found I needed to create an agents.txt in ./functions with the names of the agents?
* Additionally the agent index.yaml needed to be placed in functions/agents/<myagent>/index.yaml?
* the rag and default directory was however ./agents/<myagent>/ ?
* The --info seems to show other configaration directories but is not complete.


It took me quite a bit of time to get this to work. It seems very useful to create these agents but without a tiny bit of documentation in a Agent Guide I think many people will give up.
